smack-experimental/src/main/java/org/jivesoftware/smackx/spoiler/provider/SpoilerProvider.java

```diff
@@ -0,0 +1,53 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler.provider;
+
+import static org.xmlpull.v1.XmlPullParser.END_TAG;
+import static org.xmlpull.v1.XmlPullParser.TEXT;
+
+import org.jivesoftware.smack.provider.ExtensionElementProvider;
+import org.jivesoftware.smackx.spoiler.element.SpoilerElement;
+
+import org.xmlpull.v1.XmlPullParser;
+
+public class SpoilerProvider extends ExtensionElementProvider<SpoilerElement> {
+
+    @Override
+    public SpoilerElement parse(XmlPullParser parser, int initialDepth) throws Exception {
+        String lang = null;
+        String hint = null;
+
+        // TODO: Find a better way to get 'xml:lang' attribute value
```

Comment:
See `ParserUtils.getXmlLang()` :)

Reply:
Ah, I searched in XmlPullParser for it, since the corresponding setter is in XmlStringBuilder.